### PR TITLE
Change IsSynchronized and SyncRoot of CryptographicAttributeObjectCollection class from explicitly implementing the interface to implicitly implementing the interface.

### LINKED ISSUE #29607: CryptographicAttributeObjectCollection public API breaking change
Hello,
I need to reuse corefx System.Security.Cryptography sources in Mono. Using [API Diff tool](https://jenkins.mono-project.com/job/test-mono-pull-request-amd64/14156//Public_API_Diff/) I see some  public API breaking change in [CryptographicAttributeObjectCollection](https://github.com/dotnet/corefx/blob/master/src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/CryptographicAttributeObjectCollection.cs#L145-L159).
Could you please help me find out why this change was introduced?
Related to https://github.com/mono/mono/pull/8629

### PR DESCRIPTION
**Pull request by MaximLipnin**
Fixes #29607

### CODE REVIEW

File: src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/CryptographicAttributeObjectCollection.cs [hunk 1]
```diff
@@ -102,6 +102,22 @@ public int Count
             }
         }
 
+        public bool IsSynchronized
+        {
+            get
+            {
+                return ((ICollection)_list).IsSynchronized;
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/CryptographicAttributeObjectCollection.cs]: Why did you change this from `return false;`?

Comment (MaximLipnin) [reply src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/CryptographicAttributeObjectCollection.cs]: List should return `false` by default here. But I can change it to return false explicitly.

Comment (stephentoub) [reply src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/CryptographicAttributeObjectCollection.cs]: Yes, please revert.  There's no value in paying for the extra call when it's always going to return false.

Comment (MaximLipnin) [reply src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/CryptographicAttributeObjectCollection.cs]: addressed

File: src/System.Security.Cryptography.Pkcs/src/MatchingRefApiCompatBaseline.netstandard.txt [hunk 2]
```diff
@@ -1,3 +1,6 @@
 # These members don't belong in netstandard reference because of type unification with netfx (the members are new in net472).
 MembersMustExist : Member 'System.Security.Cryptography.Pkcs.SignerInfo.GetSignature()' does not exist in the reference but it does exist in the implementation.
 MembersMustExist : Member 'System.Security.Cryptography.Pkcs.SignerInfo.SignatureAlgorithm.get()' does not exist in the reference but it does exist in the implementation.
+
+MembersMustExist : Member 'System.Security.Cryptography.CryptographicAttributeObjectCollection.IsSynchronized.get()' does not exist in the reference but it does exist in the implementation.
+MembersMustExist : Member 'System.Security.Cryptography.CryptographicAttributeObjectCollection.SyncRoot.get()' does not exist in the reference but it does exist in the implementation.
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Pkcs/src/MatchingRefApiCompatBaseline.netstandard.txt]: This seems off.  This is just comparing the implementation to the netstandard build of the ref, but you changed the ref.

So it feels like changes to this file shouldn't be necessary...
